### PR TITLE
Move column format logic to `initialize` method

### PR DESCRIPTION
**Pull request by jasonyanwenl**
This PR fixes the issue that `OntonotesReader` does not accept `BaseReader` level init arguments. And this is partially related to this [issue](https://github.com/asyml/forte/issues/334)

### Description of changes
Move initialization logic to `initialize` method. And move default `column format` to `default_config`.

### Possible influences of this PR.
Another PR (main train SRL which will be raised soon) will use this feature.

### Test Conducted
The change should not affect current unit testings

### CODE REVIEW

File: forte/data/readers/ontonotes_reader.py [hunk 1]
```diff
@@ -117,6 +126,15 @@ def __init__(self, column_format: Optional[List[Optional[str]]] = None):
             if field not in seen_fields:
                 raise ValueError(f"'{field}' field is required")
 
+    @classmethod
+    def default_configs(cls):
+        config: dict = super().default_configs()
+
+        config.update({
+            "column_format": cls._DEFAULT_FORMAT
```
Comment (hunterhector) [inline forte/data/readers/ontonotes_reader.py:134]: it would be helpful to add the description of the column format as the docstring for this class.

**Review by hunterhector — COMMENTED** [review]
The fix should be OK. A small docstring modification would be nice.

File: forte/data/readers/ontonotes_reader.py (hunk 1 above)
Comment (jasonyanwenl) [reply forte/data/readers/ontonotes_reader.py:167]: Done

**Review by hunterhector — APPROVED** [review]
lgtm

**Pull request merged**